src/app/products/_components/detail/ProductDetailClient.tsx

```diff
@@ -0,0 +1,193 @@
+'use client';
+
+import { useState } from 'react';
+import type { IProduct, IProductOptionDetail, IProductOptions } from '@/api/product';
+import ProductDegtailCards from './ProductDetailCards';
+import ProductDetailSelectOptions from './ProductDetailSelectOptions';
+import { useRouter } from 'next/navigation';
+
+export interface ISelectOptionDetail {
+  count: number;
+  optionName: string;
+  value: string;
+  quantity: number;
+  price: number;
+}
+
+const ProductDetailClient: React.FC<{ product: IProduct }> = ({ product }) => {
+  const router = useRouter();
+  const [selectedTab, setSelectedTab] = useState('상세정보');
+  const [selectOptionDetails, setSelectOptionDetails] = useState<ISelectOptionDetail[]>([]);
+
+  // 대표 이미지 URL 찾기
+  const representativeImageUrl = product.options.flatMap((option) => option.optionDetails).find((detail) => detail.url);
+
+  function handleAddOptionsDetail(option: IProductOptions, detail: IProductOptionDetail) {
+    const find = selectOptionDetails.find((item) => item.value === detail.value);
+    if (find) {
+      setSelectOptionDetails([
+        ...selectOptionDetails.map((item) => {
+          if (item.value === detail.value) return { ...item, count: item.count + 1 };
+          return item;
+        }),
+      ]);
+      // TODO: 갯수 추가하는 코드 필요
+    } else {
+      setSelectOptionDetails([
+        ...selectOptionDetails,
+        {
+          count: 1,
+          optionName: option.name,
+          value: detail.value,
+          quantity: detail.quantity,
+          price: product.price,
+        },
+      ]);
+    }
+  }
+
+  function handleRemoveOption(option: ISelectOptionDetail) {
+    setSelectOptionDetails([...selectOptionDetails.filter((item) => item.value !== option.value)]);
+  }
+
+  function handleOptionCount(option: ISelectOptionDetail, flag: boolean) {
+    setSelectOptionDetails([
+      ...selectOptionDetails.map((item) => {
+        if (item.value === option.value)
+          return { ...item, count: flag ? item.count + 1 : item.count - 1 < 1 ? 1 : item.count - 1 };
+        return item;
+      }),
+    ]);
+  }
+
+  const handlePurchase = () => {
+    if (selectOptionDetails.length > 0) {
+      // JSON 데이터를 URL에 포함할 수 있도록 인코딩
+      const encodedData = encodeURIComponent(JSON.stringify(selectOptionDetails));
+      // 'test' 페이지로 이동하면서 데이터 전달
+      router.push(`/test?data=${encodedData}`);
```

Comment:
```suggestion
      // JSON 데이터를 URL에 포함할 수 있도록 인코딩
      const paramData = {
        product,
        selectOptionDetails
      }
      const encodedData = encodeURIComponent(JSON.stringify(paramData));
      // 'test' 페이지로 이동하면서 데이터 전달
      router.push(`/test?data=${encodedData}`);
```

지훈님!! 어제 이렇게 고친걸로 param data 가져와서 작업했습니다!! 통합할 때 이렇게 바꾸면 될거같아요!